### DISCUSSION TITLE
Json field is case sensitive

### DISCUSSION THREAD

**Discussion by sajt** [filamentphp/filament#2162] [Help] [answered]
I'm using Filament with Spatie translatable plugin. I have some json field of course. I made these fields searchable in my resource file. This is my model:

```php
        Schema::create('articles', function (Blueprint $table) {
            $table->id();
            $table->json('title');
            $table->json('subtitle');
            $table->json('body');
            $table->json('slug');
            $table->boolean('published')->default(false);
            $table->timestamps();
        });
```

And this is the part of the resource

```php
public static function table(Table $table): Table
    {
        storage_path('/public/storage/');
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('title')->sortable()->searchable(),
                Tables\Columns\TextColumn::make('subtitle'),
                Tables\Columns\BooleanColumn::make('published'),
                Tables\Columns\TextColumn::make('created_at')->sortable()
                    ->dateTime(),
                Tables\Columns\TextColumn::make('updated_at')->sortable()
                    ->dateTime(),
            ])
            ->filters([
                //
            ]);
    }
```
When I try to search for `STRABAG' string the query looks like (the query string is lowercased): 
```sql
SELECT count(*) as aggregate FROM `articles` WHERE (json_unquote(json_extract(`title`, '$."en"')) like '%strabag%')
```
It's result is 0, but I have a field with STRABAG. 

How can I solve this problem?

**Comment by danharrin** [comment, marked as answer]
Fixed by #2183

**Comment by dev3k** [comment]
sorry to post on old discussion but it seems v3.0.93 has the same issue

**Reply by dev3k** [reply]
nvm, seems can be fixed with ->forceSearchCaseInsensitive(), sorry again

**Reply by danharrin** [reply]
Are you on Postgres then?

**Reply by dev3k** [reply]
no, Mysql 8.1

**Reply by danharrin** [reply]
Sounds like a bug to be honest

Forcing case insensitivity should be defaulted to true if you're using JSON or something

**Reply by dev3k** [reply]
I agree. I use the JSON column type, and it's always case-sensitive. Maybe just a note in the docs would be enough?

**Reply by danharrin** [reply]
I think we can detect it and do it automatically. Can you please do me a favour and open an issue with a reproduction repo where I can fix it to be automatic?

**Reply by dev3k** [reply]
sure thing

**Reply by danharrin** [reply]
thanks, cant guarantee when i will get to it, but at least its in my backlog

**Reply by dev3k** [reply]
While building the reproduction repo, I faced a bug  #7537 , you can check my repo https://github.com/dev3k/json-column for that bug , wanted to write it there but it got locked 😅

**Reply by danharrin** [reply]
Can you open a quick issue that just tells me which page to visit etc?

**Reply by dev3k** [reply]
done #9649 , will add the case sensitivity later on, once that issue get fixed 

thanks for your hard work 👍

**Reply by danharrin** [reply]
thanks!

**Reply by danharrin** [reply]
all fixed now :)

**Reply by dev3k** [reply]
thanks 👍 ,  will open new issue for json field case sensitivity

**Reply by dev3k** [reply]
#9736
